File: C++/563. Binary Tree Tilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int ans;
    int dfs(TreeNode* node)
    {
        if(node == nullptr)
            return 0;
        int l = dfs(node->left);
        int r = dfs(node->right);
        ans += abs(l-r);
        return l + r + node->val;
    }
    int findTilt(TreeNode* root) {
        ans = 0;
        dfs(root);
        return ans;
    }
};

File: Python/563. Binary Tree Tilt.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def findTilt(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        ans = [0]
        def dfs(node):
            if node is None:
                return 0
            l = dfs(node.left)
            r = dfs(node.right)
            ans[0] += abs(l-r)
            return l + r + node.val
        dfs(root)
        return ans[0]
            